python/cudf/cudf/core/column/timedelta.py

```diff
@@ -125,97 +151,105 @@ def to_pandas(
-            if isinstance(rhs, cudf.Scalar):
-                if rhs.is_valid():
-                    rhs = rhs.value.astype(common_dtype).astype("float64")
+        self, other: ColumnBinaryOperand, op: str
+    ) -> Tuple["column.ColumnBase", ColumnBinaryOperand, DtypeObj]:
+        this: ColumnBase = self
+        if pd.api.types.is_timedelta64_dtype(other.dtype):
+            common_dtype = determine_out_dtype(self.dtype, other.dtype)
+            this = self.astype(common_dtype).astype("float64")
+            if isinstance(other, cudf.Scalar):
+                if other.is_valid():
+                    other = other.value.astype(common_dtype).astype("float64")
                 else:
-                    rhs = cudf.Scalar(None, "float64")
+                    other = cudf.Scalar(None, "float64")
             else:
-                rhs = rhs.astype(common_dtype).astype("float64")
+                other = other.astype(common_dtype).astype("float64")
 
-            out_dtype = cudf.dtype("float64" if op == "truediv" else "int64")
-        elif rhs.dtype.kind in ("f", "i", "u"):
+            out_dtype = cudf.dtype(
+                "float64" if op == "__truediv__" else "int64"
+            )
+        elif other.dtype.kind in ("f", "i", "u"):
             out_dtype = self.dtype
         else:
             raise TypeError(
-                f"Division of {self.dtype} with {rhs.dtype} "
+                f"Division of {self.dtype} with {other.dtype} "
                 f"cannot be performed."
             )
 
-        return lhs, rhs, out_dtype
+        return this, other, out_dtype
 
-    def binary_operator(
-        self, op: str, rhs: BinaryOperand, reflect: bool = False
+    def _binaryop(
+        self, other: ColumnBinaryOperand, op: str
     ) -> "column.ColumnBase":
-        rhs = self._wrap_binop_normalization(rhs)
-        lhs, rhs = self, rhs
-
-        if op in {"eq", "ne", "lt", "gt", "le", "ge", "NULL_EQUALS"}:
-            out_dtype = self._binary_op_lt_gt_le_ge_eq_ne(rhs)
-        elif op == "mul":
-            out_dtype = self._binary_op_mul(rhs)
-        elif op == "mod":
-            out_dtype = self._binary_op_mod(rhs)
-        elif op in {"truediv", "floordiv"}:
-            lhs, rhs, out_dtype = self._binary_op_div(rhs, op)  # type: ignore
-            op = "truediv"
-        elif op == "add":
-            out_dtype = _timedelta_add_result_dtype(lhs, rhs)
-        elif op == "sub":
-            out_dtype = _timedelta_sub_result_dtype(lhs, rhs)
+        reflect, op = self._check_reflected_op(op)
+        if (other := self._wrap_binop_normalization(other)) is NotImplemented:
+            return NotImplemented
+
+        this: ColumnBinaryOperand = self
+        if op in {
+            "__eq__",
+            "__ne__",
+            "__lt__",
+            "__gt__",
+            "__le__",
+            "__ge__",
+            "NULL_EQUALS",
+        }:
+            out_dtype = self._binary_op_lt_gt_le_ge_eq_ne(other)
```

Comment:
I'd suggest a different function name here. It doesn't include `NULL_EQUALS` and is pretty long / confusing.
```suggestion
            out_dtype = self._binary_op_comparison(other)
```

Reply:
I'm going to push this to a future PR as well. I'm hoping to improve the per-column binary operation logic in a number of places, but I'm trying to restrict this PR to _just_ sorting out the public API so that we can decouple future changes to the Frame layer from changes to the Column layer.